tsfresh/feature_extraction/feature_calculators.py

```diff
@@ -762,11 +766,20 @@ def number_peaks(x, n):
     :return: the value of this feature
     :return type: float
     """
-    res = list()
+    x = np.asarray(x)
+    x_reduced = x[n:-n]
+
+    res = None
     for i in range(1, n + 1):
-        res += [(x > np.roll(x, i))[n:-n]]
-        res += [(x > np.roll(x, -i))[n:-n]]
-    return sum(reduce(lambda a, b: a & b, res))
+        result_first = (x_reduced > np.roll(x, i)[n:-n])
+
+        if res is None:
```

Comment:
No, because you need to do a &= with it and 0 has the size of 1 (a single value) whereas result_first has a finite length (an array). This will break.